ios/RNSentry.m

```diff
@@ -270,47 +270,34 @@ - (void)setEventEnvironmentTag:(SentryEvent *)event
               });
 }
 
-RCT_EXPORT_METHOD(captureEnvelope:(NSDictionary * _Nonnull)envelopeDict
+RCT_EXPORT_METHOD(captureEnvelope:(NSArray * _Nonnull)bytes
+                  options: (NSDictionary * _Nonnull)options
                   resolve:(RCTPromiseResolveBlock)resolve
                   rejecter:(RCTPromiseRejectBlock)reject)
 {
-    if ([NSJSONSerialization isValidJSONObject:envelopeDict]) {
-        SentrySdkInfo *sdkInfo = [[SentrySdkInfo alloc] initWithDict:envelopeDict[@"header"]];
-        SentryId *eventId = [[SentryId alloc] initWithUUIDString:envelopeDict[@"header"][@"event_id"]];
-        SentryTraceContext *traceContext = [[SentryTraceContext alloc] initWithDict:envelopeDict[@"header"][@"trace"]];
-        SentryEnvelopeHeader *envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId sdkInfo:sdkInfo traceContext:traceContext];
-
-        NSError *error;
-        NSData *envelopeItemData = [NSJSONSerialization dataWithJSONObject:envelopeDict[@"payload"] options:0 error:&error];
-        if (nil != error) {
-            reject(@"SentryReactNative", @"Cannot serialize event", error);
-        }
-
-        NSString *itemType = envelopeDict[@"payload"][@"type"];
-        if (itemType == nil) {
-            // Default to event type.
-            itemType = @"event";
-        }
-
-        SentryEnvelopeItemHeader *envelopeItemHeader = [[SentryEnvelopeItemHeader alloc] initWithType:itemType length:envelopeItemData.length];
-        SentryEnvelopeItem *envelopeItem = [[SentryEnvelopeItem alloc] initWithHeader:envelopeItemHeader data:envelopeItemData];
+    NSMutableData *data = [[NSMutableData alloc] initWithCapacity: [bytes count]];
+    for(NSNumber *number in bytes) {
+        char byte = [number charValue];
+        [data appendBytes: &byte length: 1];
+    }
 
-        SentryEnvelope *envelope = [[SentryEnvelope alloc] initWithHeader:envelopeHeader singleItem:envelopeItem];
+    SentryEnvelope *envelope = [PrivateSentrySDKOnly envelopeWithData:data];
+    if (envelope == nil) {
+        reject(@"SentryReactNative",@"Failed to parse envelope from byte array.", nil);
+        return;
+    }
 
-        #if DEBUG
+    #if DEBUG
+        [PrivateSentrySDKOnly captureEnvelope:envelope];
+    #else
+        if (options[@'store']) {
+            // Storing to disk happens asynchronously with captureEnvelope
+            [PrivateSentrySDKOnly storeEnvelope:envelope];
+        } else {
```

Comment:
I wasn't sure how to remove this if statement. It stores the envelope for fatal errors. Or is it okay for now?

Reply:
for iOS, this is needed, iOS sends events async and if the App crashes right away, we only have time to write to the disk.

Reply:
ok, thanks.